lalrpop/src/grammar/parse_tree.rs

```diff
@@ -149,10 +149,12 @@ pub struct NonterminalData {
     pub alternatives: Vec<Alternative>
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Annotation {
     pub id_span: Span,
     pub id: InternedString,
+    // TODO nixpulvis: This is not exactly the correct type.
+    pub parameters: Vec<(InternedString, InternedString)>,
```

Comment:
I'm also not sure `InternedString` is the right choice anymore, as things like `1` used in the precedence example don't get parsed. What kinds of things would we expect for for the `value` fields of this structure.